test/e2e/run/runner.go

```diff
@@ -0,0 +1,157 @@
+package run
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/go-logr/logr"
+
+	"github.com/aws/eks-hybrid/test/e2e"
+	"github.com/aws/eks-hybrid/test/e2e/cluster"
+)
+
+type E2ERunner struct {
+	AddTestPhase    func(name, message string, err error) Phase
+	AwsCfg          aws.Config
+	Logger          logr.Logger
+	NoColor         bool
+	Paths           E2EPaths
+	TestLabelFilter string
+	TestProcs       int
+	TestTimeout     string
+	TestResources   cluster.TestResources
+	SkipCleanup     bool
+	SkippedTests    string
+}
+
+// Run runs the E2E tests and returns the failure phase with the error
+func (e *E2ERunner) Run(ctx context.Context) error {
+	// After this point, return err so that the defer cleanup can combine all potential
+	// errors into what is finally returned
+	var err error
+	defer func() {
+		if e.SkipCleanup {
+			e.Logger.Info("Skipping cluster and infrastructure cleanup via stack deletion")
+			return
+		}
+		cleaner := E2ECleanup{
+			AwsCfg:        e.AwsCfg,
+			Logger:        e.newFileLogger(e.Paths.CleanupLog),
+			TestResources: e.TestResources,
+		}
+		cleanupErr := cleaner.Run(ctx)
+		cleanupPhase := e.AddTestPhase(phaseNameCleanupCluster, "cleaning up cluster: %w", cleanupErr)
```

Comment:
not against this, but would it be cleaner if we just return the phases as a result of this Run method? and then in the top level caller you can add them to your slice